### PR TITLE
Wrap jackson exception on malformed json string

### LINKED ISSUE #114142: _search payload resulting in JsonParseException
The following error has been observed as a result of a _search call, resulting in HTTP 500 response.

```
com.fasterxml.jackson.core.JsonParseException: Unrecognized character escape '*' (code 42)
 at [Source: (org.elasticsearch.common.bytes.BytesReferenceStreamInput); line: 1, column: 40]
	at com.fasterxml.jackson.core@2.17.2/com.fasterxml.jackson.core.JsonParser._constructReadException(JsonParser.java:2648)
	at com.fasterxml.jackson.core@2.17.2/com.fasterxml.jackson.core.base.ParserBase._handleUnrecognizedCharacterEscape(ParserBase.java:1376)
	at com.fasterxml.jackson.core@2.17.2/com.fasterxml.jackson.core.json.UTF8StreamJsonParser._decodeEscaped(UTF8StreamJsonParser.java:3377)
	at com.fasterxml.jackson.core@2.17.2/com.fasterxml.jackson.core.json.UTF8StreamJsonParser._finishString2(UTF8StreamJsonParser.java:2565)
	at com.fasterxml.jackson.core@2.17.2/com.fasterxml.jackson.core.json.UTF8StreamJsonParser._finishAndReturnString(UTF8StreamJsonParser.java:2520)
	at com.fasterxml.jackson.core@2.17.2/com.fasterxml.jackson.core.json.UTF8StreamJsonParser.getText(UTF8StreamJsonParser.java:294)
	at org.elasticsearch.xcontent.impl@9.0.0/org.elasticsearch.xcontent.provider.json.JsonXContentParser.text(JsonXContentParser.java:111)
	at org.elasticsearch.xcontent@9.0.0/org.elasticsearch.xcontent.FilterXContentParser.text(FilterXContentParser.java:95)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.index.query.QueryStringQueryBuilder.fromXContent(QueryStringQueryBuilder.java:691)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.search.SearchModule.lambda$registerQuery$24(SearchModule.java:1287)
	at org.elasticsearch.xcontent@9.0.0/org.elasticsearch.xcontent.NamedXContentRegistry.parseNamedObject(NamedXContentRegistry.java:149)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.index.query.AbstractQueryBuilder$1.namedObject(AbstractQueryBuilder.java:403)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.index.query.AbstractQueryBuilder.parseInnerQueryBuilder(AbstractQueryBuilder.java:437)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.index.query.AbstractQueryBuilder.parseTopLevelQuery(AbstractQueryBuilder.java:411)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.search.builder.SearchSourceBuilder.parseXContent(SearchSourceBuilder.java:1418)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.search.builder.SearchSourceBuilder.parseXContent(SearchSourceBuilder.java:1299)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.rest.action.search.RestSearchAction.parseSearchRequest(RestSearchAction.java:182)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.rest.action.search.RestSearchAction.lambda$prepareRequest$1(RestSearchAction.java:121)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.rest.RestRequest.withContentOrSourceParamParserOrNull(RestRequest.java:579)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.rest.action.search.RestSearchAction.prepareRequest(RestSearchAction.java:120)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.rest.BaseRestHandler.handleRequest(BaseRestHandler.java:104)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.rest.RestController$1.onResponse(RestController.java:501)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.rest.RestController$1.onResponse(RestController.java:495)
	at org.elasticsearch.security@9.0.0/org.elasticsearch.xpack.security.rest.SecurityRestFilter.doHandleRequest(SecurityRestFilter.java:89)
	at org.elasticsearch.security@9.0.0/org.elasticsearch.xpack.security.rest.SecurityRestFilter.lambda$intercept$0(SecurityRestFilter.java:81)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:257)
	at org.elasticsearch.security@9.0.0/org.elasticsearch.xpack.security.authc.support.SecondaryAuthenticator.lambda$authenticateAndAttachToContext$3(SecondaryAuthenticator.java:99)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:247)
	at org.elasticsearch.security@9.0.0/org.elasticsearch.xpack.security.authc.support.SecondaryAuthenticator.authenticate(SecondaryAuthenticator.java:109)
	at org.elasticsearch.security@9.0.0/org.elasticsearch.xpack.security.authc.support.SecondaryAuthenticator.authenticateAndAttachToContext(SecondaryAuthenticator.java:90)
	at org.elasticsearch.security@9.0.0/org.elasticsearch.xpack.security.rest.SecurityRestFilter.intercept(SecurityRestFilter.java:75)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.rest.RestController.dispatchRequest(RestController.java:495)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.rest.RestController.tryAllHandlers(RestController.java:659)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.rest.RestController.dispatchRequest(RestController.java:340)
	at org.elasticsearch.serverless.rest@9.0.0/co.elastic.elasticsearch.serverless.rest.ServerlessRestController.dispatchRequest(ServerlessRestController.java:133)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.http.AbstractHttpServerTransport.dispatchRequest(AbstractHttpServerTransport.java:488)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.http.AbstractHttpServerTransport.handleIncomingRequest(AbstractHttpServerTransport.java:584)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.http.AbstractHttpServerTransport.incomingRequest(AbstractHttpServerTransport.java:461)
	at org.elasticsearch.transport.netty4@9.0.0/org.elasticsearch.http.netty4.Netty4HttpPipeliningHandler.handlePipelinedRequest(Netty4HttpPipeliningHandler.java:166)
	at org.elasticsearch.transport.netty4@9.0.0/org.elasticsearch.http.netty4.Netty4HttpPipeliningHandler.channelRead(Netty4HttpPipeliningHandler.java:145)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.codec@4.1.109.Final/io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at org.elasticsearch.transport.netty4@9.0.0/org.elasticsearch.http.netty4.Netty4HttpAggregator.channelRead(Netty4HttpAggregator.java:52)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.codec@4.1.109.Final/io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at org.elasticsearch.transport.netty4@9.0.0/org.elasticsearch.http.netty4.Netty4HttpHeaderValidator.channelRead(Netty4HttpHeaderValidator.java:74)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.codec@4.1.109.Final/io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346)
	at io.netty.codec@4.1.109.Final/io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.codec@4.1.109.Final/io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.handler@4.1.109.Final/io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1475)
	at io.netty.handler@4.1.109.Final/io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1338)
	at io.netty.handler@4.1.109.Final/io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1387)
	at io.netty.codec@4.1.109.Final/io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:530)
	at io.netty.codec@4.1.109.Final/io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:469)
	at io.netty.codec@4.1.109.Final/io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:290)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.transport@4.1.109.Final/io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.transport@4.1.109.Final/io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.transport@4.1.109.Final/io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
	at io.netty.transport@4.1.109.Final/io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
	at io.netty.transport@4.1.109.Final/io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:689)
	at io.netty.transport@4.1.109.Final/io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:652)
	at io.netty.transport@4.1.109.Final/io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.common@4.1.109.Final/io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.common@4.1.109.Final/io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at java.base/java.lang.Thread.run(Thread.java:1570)
```

this happened in context similar to #113867 .

### PR DESCRIPTION
**Pull request by henriquepaes1**
This PR closes #114142, which observed a 500 response when sending a malformed query to the _search endpoint. I identified that the problem was due to an unchecked exception thrown by the Jackson JSON library when using the .getText( ) method. The solution was to catch the JsonParseException and throw an IllegalArgumentException that gives better information on where the error happened. Now the response looks like the image below:

image

### CODE REVIEW

**Comment by rjernst** [comment]
Thanks for the change @henriquepaes1. Can you please adhere to the formatting style used by the rest of the code? Spaces between braces/parens and catch on same line as braces.

It would also be nice to have a test. It could go in `JsonXContentTests`

**Comment by henriquepaes1** [comment]
Hey @rjernst! Thanks for pointing these out! I'm struggling with the test. Should I write the test for the JsonXContent class or for the AbstractQueryBuilder, where the exception was observed in the issue? Also, should I squash the commits when I finish?

**Comment by rjernst** [comment]
> Should I write the test for the JsonXContent class or for the AbstractQueryBuilder

You could add the test to BaseXContentTestCase, which will be run for all xcontent types.

> should I squash the commits when I finish?

Don't squash. I suggest you read https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md.

**Comment by henriquepaes1** [comment]
Hey @rjernst! Are there any other things that I should correct in this PR?

**Comment by rjernst** [comment]
@elasticmachine test this please

**Comment by rjernst** [comment]
@elasticsearchmachine update branch

**Comment by rjernst** [comment]
@elasticmachine update branch

**Comment by rjernst** [comment]
@elasticmachine test this please

**Comment by rjernst** [comment]
@elasticmachine test this please

**Comment by rjernst** [comment]
@elasticmachine test this please

**Comment by rjernst** [comment]
@henriquepaes1 The test failure is in BulkRestIT, due to the change. It looks like a simple update is needed to the assertion conditions since it is no longer exposing a json_parse_exception.

```
Expected: a string containing "could not parse bulk request body"	
         but: was "method [POST], host [http://127.0.0.1:16633/], URI [/_bulk], status line [HTTP/1.1 400 Bad Request]	
    {"error":{"root_cause":[{"type":"x_content_parse_exception","reason":"[1:23] Invalid UTF-8 start byte 0xfe\n at [Source: (byte[])\"{\"create\":{\"_index\":\"����\",\"_id\":\"1\"}}\"; line: 1, column: 23]"}],"type":"x_content_parse_exception","reason":"[1:23] Invalid UTF-8 start byte 0xfe\n at [Source: (byte[])\"{\"create\":{\"_index\":\"����\",\"_id\":\"1\"}}\"; line: 1, column: 23]","caused_by":{"type":"json_parse_exception","reason":"Invalid UTF-8 start byte 0xfe\n at [Source: (byte[])\"{\"create\":{\"_index\":\"����\",\"_id\":\"1\"}}\"; line: 1, column: 23]"}},"status":400}"
```

**Comment by henriquepaes1** [comment]
Hey @rjernst! Thanks for the heads up. Just updated the test. Was there a way that I could catch this in my local environment? I haven't been able to run some tests locally.

**Comment by rjernst** [comment]
@elasticmachine test this please

**Comment by rjernst** [comment]
> Was there a way that I could catch this in my local environment?

You can, though this test is in `qa` which can be quite heavy. If you have the patience, running `./gradlew check` would have tripped this failure.

**Comment by henriquepaes1** [comment]
> > Was there a way that I could catch this in my local environment?
> 
> You can, though this test is in `qa` which can be quite heavy. If you have the patience, running `./gradlew check` would have tripped this failure.

I see. I don't know why but when I run gradlew check I get a git-related error. Idk if that's because I'm using Windows, still have to figure it out.

**Comment by henriquepaes1** [comment]
Hey @rjernst! Is this failing action also related to an integration test that relied on the previous thrown exception? Let me know if there's any action that I should complete in this PR.

**Comment by prdoyle** [comment]
@elasticmachine update branch

**Comment by prdoyle** [comment]
@elasticmachine test this please

**Comment by henriquepaes1** [comment]
>  at first glance, it looks like a failure unrelated to your change, so I've merged `main` and kicked off the tests again, hoping for an upstream fix. 🤞

Sure @prdoyle! Thanks for the heads up. Let me know if I need to fix something :)

File: libs/x-content/impl/src/main/java/org/elasticsearch/xcontent/provider/json/JsonXContentParser.java [hunk 1]
```diff
@@ -108,7 +108,13 @@ public String text() throws IOException {
...
             throwOnNoText();
         }
-        return parser.getText();
+        String text = "";
+        try {
+            text = parser.getText();
+        } catch (JsonParseException e) {
+            throw newXContentParseException(e);
+        }
+        return text;
```
Comment (prdoyle) [inline libs/x-content/impl/src/main/java/org/elasticsearch/xcontent/provider/json/JsonXContentParser.java:117]:
```suggestion
        try {
            return parser.getText();
        } catch (JsonParseException e) {
            throw newXContentParseException(e);
        }
```

**Review by prdoyle — APPROVED** [review]

File: libs/x-content/impl/src/main/java/org/elasticsearch/xcontent/provider/json/JsonXContentParser.java (hunk 1 above)
Comment (prdoyle) [reply libs/x-content/impl/src/main/java/org/elasticsearch/xcontent/provider/json/JsonXContentParser.java:117]: Yes I think so, unless you object to it. 😅

**Comment by prdoyle** [comment]
> Don't squash. I suggest you read https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md.

@rjernst - FWIW that linked document says this:

> As a final step before merging we will either ask you to squash all commits yourself or we'll do it for you.

To be clear @henriquepaes1 - we'll do it for you. 😄

**Comment by prdoyle** [comment]
@henriquepaes1 - I'd like to check with Ryan one more time before proceeding to merge this. That means it will have to wait till next week.

**Comment by henriquepaes1** [comment]
> I'd like to check with Ryan one more time before proceeding to merge this. That means it will have to wait till next week.

No problems! Thanks for the clarification and for the help in getting this PR going :)

**Comment by prdoyle** [comment]
@elasticmachine test this please

**Comment by rjernst** [comment]
@elasticmachine update branch

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Review by rjernst — APPROVED** [review]
Thanks @henriquepaes1 , LGTM

**Comment by benwtrent** [comment]
🎉🎉🎉 @henriquepaes1 🎉🎉🎉

🎉🎉🎉 Thank you!!!!! 🎉🎉🎉

**Comment by henriquepaes1** [comment]
Thanks for the guidance @benwtrent @rjernst! I'm really glad to contribute, hopefully it was the first of many!

**Pull request merged**